soc/nxp/mcx/mcxw/Kconfig.soc

```diff
@@ -23,16 +23,30 @@ config SOC_MCXW727C_CPU1
 	bool
 	select SOC_MCXW727C
 
+config SOC_MCXW236
+	bool
+	select SOC_FAMILY_MCXW
+
+config SOC_MCXW235
+	bool
+	select SOC_FAMILY_MCXW
+
 config SOC
 	default "mcxw716c" if SOC_MCXW716C
 	default "mcxw727c" if SOC_MCXW727C
+	default "mcxw236" if SOC_MCXW236
+	default "mcxw235" if SOC_MCXW235
```

Comment:
Look at `imxrt` for reference. I would retain 
```
config SOC_FAMILY_MCXW
	bool

config SOC_FAMILY
	default "mcxw" if SOC_FAMILY_MCXW
```
Everything else should go into the respective sub-folder's `Kconfig.soc` file